src/utils/get-deep-value.ts

```diff
@@ -0,0 +1,21 @@
+import { Context } from "../types";
+/**
+ * This function is a utility that allows us to access deeply nested properties in an object
+ * primarily for use with the context.payload object. It should not be overused and the developer
+ * should be aware of the potential performance implications of using this function.
+ *
+ * Example usage:
+ *
+ * - `getDeepValue(context, "payload.repository.owner.login")` will return the owner
+ * - `getDeepValue(context, ["payload", "repository", "owner", "login"])` will return the owner
+ */
+export function getDeepValue<TK extends PropertyKey, T extends Context = Context>(obj: T, path: string | string[]): TK {
```

Comment:
why is this needed?

Reply:
This was meant to be a handy generic utility fn, because with this plugin probably going to handle 10s of different webhooks eventually I thought this would be nice to have and cleaner than if...in/lots of typeguards for one-off cases. 

Because proxyCallbacks types everything well already it'll rarely be needed but good for a fallback going from strict-type fn into loose type like `handler => addComment` because addComment needs to be `Context` when handler is `Context<some webhook> => Context`... if that makes sense.

Used rn with addCommentToIssue as a fallback for obtaining repo, owner, login from the payload when 90% of the payloads we care about contain these it's the pesky "push" and I think one more that messes it up.

If you think it's of no value I will remove

Reply:
so you're trying to avoid this:
```js
if (!("repository" in context.payload)) {
  return
}
const owner = context.payload.repository?.owner?.login;
if (!owner) {
  return
}
```
but I think `getDeepValue` reduces type safety because you have to specify the return type and can introduce bugs because you can make a typo and there won't be any warnings